File: datasets/lua-keep-incoder_6b_temp_0.2-3/HumanEval_31_is_prime.lua
-- Return true if a given number is prime, and false otherwise.
-- >>> is_prime(6)
-- False
-- >>> is_prime(101)
-- True
-- >>> is_prime(11)
-- True
-- >>> is_prime(13441)
-- True
-- >>> is_prime(61)
-- True
-- >>> is_prime(4)
-- False
-- >>> is_prime(1)
-- False
local function is_prime(n)
  -- http://primes.utm.edu/lists/small/millions/
  -- http://primes.utm.edu/lists/small/millions/test.html
  -- http://primes.utm.edu/lists/small/millions/large/
  -- http://primes.utm.edu/lists/small/millions/large/test.html
  -- http://primes.utm.edu/lists/small/millions/medium/
  -- http://primes.utm.edu/lists/small/millions/medium/test.html
  -- http://primes.utm.edu/lists/small/millions/small/
  -- http://primes.utm.edu/lists/small/millions/small/test.html
  -- http://primes.utm.edu/lists/small/millions/medium/large/
  -- http://primes.utm.edu/lists/small/millions/medium/large/test.html
  -- http://primes.utm.edu/lists/small/millions/medium/small/
  -- http://primes.utm.edu/lists/small/millions/medium/small/test.html
  -- http://primes.utm.edu/lists/small/millions/large/medium/
  -- http://primes.utm.edu/lists/small/millions/large/medium/test.html
  -- http://primes.utm.edu/lists/small/millions/large/small/
  -- http://primes.utm.edu/lists/small/millions/large/small/test.html
  -- http://primes.utm.edu/lists/small/millions/medium/medium/
  -- http://primes.utm.edu/lists/small/millions/medium/medium/test.html
  -- http://primes.utm.edu/lists/small/millions/medium/small/large/
  -- http://primes.utm.edu/lists/small/millions/medium/small/large/test.html
  -- http://primes.utm.edu/lists/small/millions/medium/small/small/
  -- http://primes.utm.edu/lists/small/millions/medium/small/small/test.html
  -- http://primes.utm.edu/lists/small/millions/large/medium/large/
  -- http://primes.utm.edu/lists/small/millions/large/medium/large/test.html
  -- http://primes.utm.edu/lists/small/millions/large

lu = require('luaunit')

function test_humaneval()
local candidate = is_prime
    lu.assertEquals(candidate(6), false)
    lu.assertEquals(candidate(101), true)
    lu.assertEquals(candidate(11), true)
    lu.assertEquals(candidate(13441), true)
    lu.assertEquals(candidate(61), true)
    lu.assertEquals(candidate(4), false)
    lu.assertEquals(candidate(1), false)
    lu.assertEquals(candidate(5), true)
    lu.assertEquals(candidate(11), true)
    lu.assertEquals(candidate(17), true)
    lu.assertEquals(candidate(85), false)
    lu.assertEquals(candidate(77), false)
    lu.assertEquals(candidate(255379), false)
end

os.exit(lu.LuaUnit.run())

File: datasets/R-keep-code_davinci_001_temp_0.2/HumanEval_31_is_prime.R
# Return true if a given number is prime, and false otherwise.
-- >>> is_prime(6)
-- False
-- >>> is_prime(101)
-- True
-- >>> is_prime(11)
-- True
-- >>> is_prime(13441)
-- True
-- >>> is_prime(61)
-- True
-- >>> is_prime(4)
-- False
-- >>> is_prime(1)
-- False
is_prime <- function(n)
{
    if (n == 1)
    {
        return(FALSE)
    }
    else if (n == 2)
    {
        return(TRUE)
    }
    else
    {
        for (i in 2:(n-1))
        {
            if (n %% i == 0)
            {
                return(FALSE)
            }
        }
        return(TRUE)
    }

test_humaneval <- function() {
candidate <- is_prime
    if(!identical(candidate(6), false)){quit('no', 1)}
    if(!identical(candidate(101), true)){quit('no', 1)}
    if(!identical(candidate(11), true)){quit('no', 1)}
    if(!identical(candidate(13441), true)){quit('no', 1)}
    if(!identical(candidate(61), true)){quit('no', 1)}
    if(!identical(candidate(4), false)){quit('no', 1)}
    if(!identical(candidate(1), false)){quit('no', 1)}
    if(!identical(candidate(5), true)){quit('no', 1)}
    if(!identical(candidate(11), true)){quit('no', 1)}
    if(!identical(candidate(17), true)){quit('no', 1)}
    if(!identical(candidate(85), false)){quit('no', 1)}
    if(!identical(candidate(77), false)){quit('no', 1)}
    if(!identical(candidate(255379), false)){quit('no', 1)}
}